### PR TITLE
Api Request Handling

### PR DESCRIPTION
**Pull request by saniazafar**
Review the Network part

### CODE REVIEW

File: JSONDemoApp/Network/NetworkManager.h [hunk 1]
```diff
@@ -8,10 +8,12 @@
 
 #import <Foundation/Foundation.h>
 
-NS_ASSUME_NONNULL_BEGIN
 
 @interface NetworkManager : NSObject
 
++ (NetworkManager *)sharedInstance;
+- (void) requestApiData: (NSString *)endPoint completionHandler:(void (^)(NSData *data, NSURLResponse *response, NSError *error))completionHandler;
```
Comment (noorulain17) [inline JSONDemoApp/Network/NetworkManager.h]: add empty line between class and instance methods

File: JSONDemoApp/Network/NetworkManager.h [hunk 2]
```diff
@@ -8,10 +8,12 @@
...
+
 @end
 
-NS_ASSUME_NONNULL_END
+
```
Comment (noorulain17) [inline JSONDemoApp/Network/NetworkManager.h:17]: remove this extra line

File: JSONDemoApp/ViewController/ViewController.m [hunk 3]
```diff
@@ -7,6 +7,7 @@
 //
 
 #import "ViewController.h"
+#import "NetworkManager.h"
```
Comment (noorulain17) [inline JSONDemoApp/ViewController/ViewController.m]: unused import?

File: JSONDemoApp/Network/NetworkManager.h (hunk 1 above)
Comment (saniazafar) [reply JSONDemoApp/Network/NetworkManager.h]: Done

File: JSONDemoApp/Network/NetworkManager.h (hunk 2 above)
Comment (saniazafar) [reply JSONDemoApp/Network/NetworkManager.h:17]: Done

File: JSONDemoApp/ViewController/ViewController.m (hunk 3 above)
Comment (saniazafar) [reply JSONDemoApp/ViewController/ViewController.m]: Yes its was used before, removed now.

File: JSONDemoApp/Constant/Constants.h [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+//
+//  Created by Sania Zafar on 12/12/18.
+//  Copyright © 2018 Sania Zafar. All rights reserved.
+//
+
+
+
+const NSString *BASE_URL = @"https://api.namshi.com/rose/u/";
+const NSString *catalog = @"men-shoes";
+const NSString *PDP = @"product/SK034SH14PAV";
```
Comment (BaluNaik) [inline JSONDemoApp/Constant/Constants.h]: @saniazafar it should be like this `const NSString *PDP = @"product/{sku}"`

**Review by BaluNaik — APPROVED** [review]

**Pull request merged**